bin/handlerFunctions.js

```diff
@@ -636,100 +666,68 @@ async function handleRawRequest(argv) {
 }
 
 // put file function
-async function push(argv) {
-    // get player data from argv
-    let playerData;
-    try {
-        playerData = await pullData(argv);
-        // playerData[0] = playerUser, [1] = playerIP, [2] = playerPW
-    } catch (err) {
-        errorHandler(err);
-        return;
-    }
+async function pushFile(file, requestOptions, playerData, isRawData, isVerbose) {
+    logIfOption(`Pushing file: ${file}`, isVerbose);
 
-    logIfOption('Pushing file(s) to ' + argv.playerName, argv.verbose);
-    logIfOption('      IP address: ' + playerData[1] + ', username: ' + playerData[0] + ', password: ' + playerData[2], argv.verbose);
-
-    let requestOptions = {
-        method: 'PUT',
-        url: 'http://' + playerData[1] + '/api/v1/files/sd/' + argv.location,
-    };
-
-    // check if file or directory
-    let path = argv.FileDirectory;
-    let absPath = currentPath.resolve(path);
-    let isFile;
+    const formData = new FormData();
+    formData.append('file', fs.createReadStream(file));
+    requestOptions.body = formData;
+    requestOptions.headers = formData.getHeaders();
 
     try {
-        isFile = await checkDir(path);
-        logIfOption('Checking if provided path is a file or directory', argv.verbose)
+        const response = await requestFetch(
+            requestOptions,
+            playerData[0],
+            playerData[2],
+            file
+        );
+        
+        if (!isRawData) {
+            console.log(`${response.data.result.results} uploaded: ${response.data.result.success}`);
+        } else if (isRawData) {
+            let resString = JSON.stringify(response.data.result);
+            logIfOption(resString, isRawData);
+        }
     } catch (err) {
         errorHandler(err);
     }
+}
 
-    if (isFile === true) {
-        logIfOption('Provided path is a file', argv.verbose)
-
-        let form = new formData();
-        let fileToUpload = fs.createReadStream(path);
-        form.append("file", fileToUpload, {filename: path});
-        requestOptions.body = form;
-        logIfOption('FormData created and appended to request body', argv.verbose)
-
-        // if file, push file
-        
+async function push(argv){
+    try {
+        const playerData = await pullData(argv); // playerData[0] = playerUser, [1] = playerIP, [2] = playerPW
```

Comment:
Do we need to add some validations for the `playerData` parameters? Since they are expected to be in the same sequence.

Reply:
that's a good idea Bhakti.

Reply:
Good point - yes, sanitization is crucial helping to minimize errors / poor UX/DX. I'll add checks here.

Reply:
Okay, approving this for now. Let's add validations as a separate ticket.

Reply:
Noted - ticket filed.